library/pool.c

```diff
@@ -55,7 +55,10 @@ static void dnet_work_pool_stop(struct dnet_work_pool_place *place)
 
 	for (i = 0; i < place->pool->num; ++i) {
 		wio = &place->pool->wio_list[i];
-		pthread_join(wio->tid, NULL);
+		if (!wio->joined) {
+			pthread_join(wio->tid, NULL);
+			wio->joined = 1;
+		}
```

Comment:
You haven't described why do you need this flag? It is a bug and it should be fixed instead of introducing this binary option, which halves cleanup path to safe and unsafe parts. Please fix cleanup problem, I'm strongly against patch with such quirks.